docs/content/guides/dagster/building-and-memoizing-versioned-assets.mdx

```diff
@@ -0,0 +1,191 @@
+---
+title: Building and memoizing versioned assets | Dagster Docs
+---
+
+# Building and memoizing versioned assets
+
+This guide demonstrates how to build memoizable graphs of assets. Memoizable assets help avoid unnecessary recomputation, speed up the developer workflow, and save computational resources.
+
+---
+
+# Context
+
+Dagster embraces functional programming principles. The core abstraction is a software-defined asset which, in its idealized form, is a side-effect-free ("pure") function that produces an output asset from input assets.
+
+Pure functions can be made more efficient with a technique called "memoization". A memoized function has an outer layer in which previously computed outputs are cached and indexed by their corresponding inputs. When a memoized function is called twice with identical inputs, the cached result can be returned on the second call.
+
+In computationally expensive data pipelining, this approach can yield tremendous benefits. Dagster offers support for memoized assets via a versioning system. Because both code and input values can affect the output of a function, Dagster has two kinds of versions:
+
+1. Code version. A string that represents the version of the code that computes an asset. This is the `code_version` argument of <PyObject object="asset" decorator="true" />.
+2. Data version. A string that represents the version of the value of the asset. This is represented as a <PyObject object="LogicalVersion" /> object.
+
+---
+
+## Step One: Understanding Staleness
+
+Dagster automatically computes the logical version of software-defined assets. It does this by hashing a code version together with the logical versions of any input assets.
+
+Let's start with a trivial asset that returns a hardcoded number:
+
+```python file=/guides/dagster/building_versioned_assets/vanilla_asset.py
+from dagster import asset
+
+
+@asset
+def a_number():
+    return 1
+```
+
+We materialize it and look at the entry in the Asset Catalog:
+
+<Image
+alt="Simple asset logical version"
+src="/images/guides/building-versioned-assets/simple-asset-in-catalog.png"
+width={4064}
+height={2488}
+/>
+
+
+You'll notice a logical version in the "System tags" section in the materialization details. This is the hash dagster has automatically computed. In this case, since there are no inputs, the logical version is a hash of the code version only, which is also visible in "System tags".
```

Comment:
> make the change to hash guide

Do you mean here that if there are no inputs, the logical version is just a copy of the `code_version`?

Also, it would be iffy to put that in this release, so do you want any changes to this section atm?

Reply:
Ok we can leave that for next week.